### PR TITLE
Chore/multiple fixes ci precommit

### PR DESCRIPTION
**Pull request by theissenhelen**
fixes/updates:
- correct downstream-ci-hpc workflow call
- ignore pre-commit-config and readthedocs for changelog updates
- updates pre-commit config

adds:

- merge strategy for CHANGELOG.md
- adds CODEOWNERS file

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -59,12 +65,21 @@ repos:
...
-  rev: "0.0.6"
-  hooks:
-  - id: optional-dependencies-all
-    args: ["--inplace", "--exclude-keys=dev,docs,tests", "--group=dev=all,docs,tests"]
 - repo: https://github.com/tox-dev/pyproject-fmt
-  rev: "2.1.3"
+  rev: "2.2.3"
   hooks:
   - id: pyproject-fmt
+-   repo: https://github.com/jshwi/docsig # Check docstrings against function sig
```
Comment (floriankrb) [inline .pre-commit-config.yaml:72]: this part breaks the build.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -20,8 +20,14 @@ repos:
   - id: no-commit-to-branch # Prevent committing to main / master
   - id: check-added-large-files # Check for large files added to git
   - id: check-merge-conflict # Check for files that contain merge conflict
+-   repo: https://github.com/pre-commit/pygrep-hooks
```
Comment (floriankrb) [inline .pre-commit-config.yaml:23]: this could be removed to fix the build.

File: .gitignore [hunk 3]
```diff
@@ -186,3 +187,12 @@ _build/
 *.sync
 _version.py
 *.code-workspace
+
```
Comment (floriankrb) [inline .gitignore:190]: let's remove duplicates?

**Comment by floriankrb** [comment]
Is this now obsolete? Should we close it?

**Comment by HCookie** [comment]
> Is this now obsolete? Should we close it?

Some of the changes here might still be useful to include. Others will need to be removed.

File: .gitignore [hunk 4]
```diff
@@ -186,3 +187,12 @@ _build/
...
+
+# Environments
+.envrc
```
Comment (floriankrb) [inline .gitignore:193]: this is really many .envrc ! ;-)

**Review by floriankrb — APPROVED** [review]
I don't know what this changes about the github actions, but this definitely doesn't break the code. So IMHO, it can be merged.

**Pull request merged**